File: Code_cpp/14_string-easy_最长公共前缀.cpp
/*
 * @lc app=leetcode.cn id=14 lang=cpp
 *
 * [14] 最长公共前缀
 */

// @lc code=start
#include <vector>
#include <string>

using namespace std;
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(!strs.size())
            return "";
        
        int matchNum = 0;
        int strLength = strs[0].size();
        for (int i = 0; i < strLength; i++,matchNum++)
        {
            char curr_char = strs[0][i];
            for(int j = 0;j<strs.size();j++)
            {
                if(matchNum == strLength || curr_char != strs[j][i])
                    return strs[0].substr(0,matchNum);
            }
        }
        return strs[0];
    }
};
// @lc code=end

File: Code_java/14_string-easy_最长公共前缀.java
/*
 * @lc app=leetcode.cn id=14 lang=java
 *
 * [14] 最长公共前缀
 */

// @lc code=start
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length == 0) {
            return "";
        }
        int strLength = strs[0].length();
        for (int i = 0; i < strLength; i++) {
            char curr_char = strs[0].charAt(i);
            for (int j = 1; j < strs.length; j++) {
                if(i == strs[j].length() || curr_char !=strs[j].charAt(i))
                    return strs[0].substring(0, i);
            }
        }
        return strs[0];

    }
}
// @lc code=end

